Adding paired-end reads

HopedWall opened the pull request.

HopedWall committed:
Changed arguments

- s2 can also be used to provide the path to the second paired-end read
- modified checkInputs accordingly

HopedWall commented on asgal:
```diff
@@ -517,6 +517,9 @@ def main():
 
     args = parser.parse_args()
 
+    #if len(args.sample1Path) == 1:
```
Another option: using -s as a list containing one or more paths to reads; this if statement would be necessary in order to keep old code working in case of a single file.

HopedWall committed:
Fixed a few if/else statements

- we have paired-end reads when: args.multiMode is false and len(samples) == 2 (or at least != 1)

HopedWall commented on asgal:
```diff
@@ -351,69 +351,80 @@ def runASGAL(args, genes, chr_genes_dict):
         anno = args.annoPath
         samples = [args.sample1Path]
+
+        # paired-end read
```
Is this condition correct?

HopedWall committed:
Improved else branch

HopedWall committed:
Added SpliceAwareAlignerPaired

- This is essentialy the same as SpliceAwareAligner, the only difference is that it takes in 2 files containing reads instead of 1

HopedWall committed:
Added paired-end version of detectEvents

- note: the actual algorithm will be implemented later

HopedWall committed:
Added STAR and Samtools commands to generate idmp

HopedWall commented on asgal:
```diff
@@ -428,6 +428,37 @@ def runASGAL(args, genes, chr_genes_dict):
         idmp = 100  # hard-coded for now
         # TODO: use STAR to align reads to genome, use SAMTOOLS to estimate the actual distance
 
+
```
Need some help on these!

HopedWall commented on asgal:
```diff
@@ -351,69 +351,80 @@ def runASGAL(args, genes, chr_genes_dict):
+        # paired-end read
+        if args.sample2Path != '-' and not args.multiMode:
+            samples += [args.sample2Path] # samples has length 2 [read1.fa, read2.fa]
+
```
Is this condition correct?

ldenti replied:
Since you are in the else branch, you are sure that `args.multiMode` is `False` so I think it's useless to check it again

HopedWall committed:
Small bugfix

- Some variables were declared multiple times in SpliceAwareAlignerPaired

HopedWall committed:
Multiple fixes to SpliceAwareAlignerPaired

- changed input parameters (was: s0 s1 now: 1 2)
- improved loop logic

HopedWall committed:
Small fix to printHelp

HopedWall committed:
Small fixes to destroy/close

HopedWall committed:
idmp estimation completed

- idmp can now be estimated based on genome+annotations+(paired-end) reads

HopedWall committed:
Added formatSAMPaired.py

- this is just the base algorithm/idea
- flags will be implemented later

HopedWall committed:
Added getFlagPaired

- Also improved loop logic

HopedWall committed:
Multiple small fixes

HopedWall committed:
Added support for unmapped reads (1)

HopedWall committed:
Added support for unmapped reads (2)

- formatSAMPaired.py

HopedWall committed:
Small fixes

HopedWall committed:
Fixed Samtools parsing errors

- from SAM Format Specification: For a unmapped paired-end or mate-pair read whose mate is mapped, the unmapped read should
have RNAME and POS identical to its mate.

HopedWall committed:
Removed useless while loops at the end

- Now that .mems files contain both mapped and unmapped reads, they are no longer necessary

HopedWall committed:
Added a few comments

- Explained how a CIGAR String is created when a read is unmapped

HopedWall committed:
Improvements to MEMs parsing (detectEventsPaired.py)

- Now obtains introns from 2 files containing MEMs instead of 1
- Now works with the new MEMs format

HopedWall committed:
Added a few stats to formatSAMPaired.py

HopedWall committed:
SpliceAwareAlignerPaired now supports placeholders

- placeholders make formatSAMPaired.py much easier by making both .mem files have the same length

HopedWall committed:
FormatSAMPaired.py now supports placeholder alignments

-still to do: make idmp match the one calculated by samtools

HopedWall committed:
Improved how tlen and idmp are calculated

- Still not correct...

HopedWall committed:
Fixed getEnd, get Tlen and getIdmp

HopedWall committed:
Small improvements to detectEventsPaired.py

- readLine now supports unmapped and placeholder mems
- added mergeIntrons

HopedWall committed:
Old version of detectEventsPaired

- ignore, just a checkpoint

HopedWall committed:
Modified getTranscriptIdmp

HopedWall committed:
Moved getTranscriptIdmp to formatSAMPaired

HopedWall committed:
Added fragment library types

- Only parsing and simple cases for now

HopedWall committed:
Fixed unstranded library types

HopedWall committed:
Fixed problem with unmapped reads/RNAME

HopedWall committed:
Fixed name ftl -> flt

HopedWall committed:
Fixed main script

- Probably not working with multiMode

HopedWall committed:
Small fixes to output format/extension

HopedWall committed:
Improved asgal main script

HopedWall committed:
Fixed small ftl bug

Nitin123-4 commented:
Can we provide paired-end fastq of bam file?
